.teamcity/_self/projects/WebApp.kt

```diff
@@ -680,6 +680,16 @@ object PreReleaseE2ETests : BuildType({
 
 	failureConditions {
 		executionTimeoutMin = 20
+		nonZeroExitCode = false
+		failOnMetricChange {
+			metric = BuildFailureOnMetric.MetricType.PASSED_TEST_COUNT
+			threshold = 50
```

Comment:
Does this mean it will fail if 50% of tests start failing? I keep forgetting exactly what these mean :P

Reply:
Ha, no kidding, they're not super intuitive!! 😆 

Technically, that would trigger a failure in that case, but this specifically is for catching when overall volume of tests suddenly plummets! It's to catch cases where someone left in `skip` or there's some pipeline error causing no tests to get run.

So if on one run, everything passes, and there are 80 passing tests, and then the next run, everything passes, but there are only 30 passing tests, it would fail!

Reply:
cool! Is there a separate mechanism which handles individual tests failing?

Reply:
To add to what Dan has said, yes this means the threshold is 50%.

Reply:
> cool! Is there a separate mechanism which handles individual tests failing?

If individual tests fail but the overall threshold is above the number in the config, then it's treated like a normal failure and so they are listed on the entry in TeamCity for the build.

The other part of this change, is that if the test that failed is one that had been previously muted in TeamCity, then we don't consider that a hard failure.